src/SamplesApp/UITests.Shared/Windows_Devices/Midi/MidiDeviceOutput.xaml.cs

```diff
@@ -0,0 +1,758 @@
+
+				case MidiMessageType.SongSelect:
+					// This list is for Songs, of which there are 128
+					PopulateParameterList(parameter1, 128, "Song");
+					break;
+
+				case MidiMessageType.SystemExclusive:
+					// Start with a clean slate
+					// Hide the first parameter
+					parameter1.Header = "";
+					parameter1.IsEnabled = false;
+					parameter1.Visibility = Visibility.Collapsed;
+					NotifyUser("Please edit the message in the textbox by clicking on 'F0 F7'");
+					break;
+
+				default:
+					// Start with a clean slate				
+					// Hide the first parameter
+					parameter1.Header = "";
+					parameter1.IsEnabled = false;
+					parameter1.Visibility = Visibility.Collapsed;
+					NotifyUser("");
+					break;
+			}
+		}
+
+		/// <summary>
+		/// React to Parameter1 selection change as appropriate
+		/// </summary>
+		/// <param name="sender">Element that fired the event</param>
+		/// <param name="e">Event arguments</param>
+		private void Parameter1_SelectionChanged(object sender, object e)
+		{
+#if WINDOWS_UWP
+			int parameter1SelectedIndex = int.Parse(this.parameter1.Text);
+#else
+			// Find the index of the user's choice
+			int parameter1SelectedIndex = (int)GetParameterValue(parameter1);
+#endif
+
+			// Some MIDI message types don't need additional parameters past parameter 1
+			// For them, show the Send button as soon as user selects parameter 1 value from the list
+			switch (_currentMessageType)
+			{
+				case MidiMessageType.SongPositionPointer:
+				case MidiMessageType.SongSelect:
+
+					if (parameter1SelectedIndex != -1)
+					{
+						sendButton.IsEnabled = true;
+					}
+					break;
+
+				default:
+					sendButton.IsEnabled = false;
+					break;
+			}
+
+			// Update the second parameter list depending on the first parameter selection
+			// If no further parameters are required, the list is emptied and hidden
+			UpdateParameterList2();
+		}
+
+		/// <summary>
+		/// For MIDI message types that need the second parameter, populate the list
+		/// based on the message type. For message types that don't need the second
+		/// parameter, empty and hide it
+		/// </summary>
+		private void UpdateParameterList2()
+		{
+#if WINDOWS_UWP
+			int parameter2SelectedIndex = int.Parse(this.parameter2.Text);
+#else
+			// Find the index of the user's choice
+			int parameter2SelectedIndex = (int)GetParameterValue(parameter2);
+#endif
+
+			// Some MIDI message types don't need additional parameters past parameter 2
+			// For them, show the Send button as soon as user selects parameter 2 value from the list
+			switch (_currentMessageType)
+			{
+				case MidiMessageType.ProgramChange:
+				case MidiMessageType.ChannelPressure:
+				case MidiMessageType.PitchBendChange:
+				case MidiMessageType.MidiTimeCode:
+
+					if (parameter2SelectedIndex != -1)
+					{
+						sendButton.IsEnabled = true;
+					}
+					break;
+
+				default:
+					sendButton.IsEnabled = false;
+					break;
+			}
+
+			switch (_currentMessageType)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [This condition was just checked on line 510.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5305560)